### PR TITLE
[SPARTA-648] Ingestion parser is not working

### PR DESCRIPTION
**Pull request by danielcsant**
Due to Avro serialization in Decision it is neccesary this upgrade

**Given** this doc https://docs.google.com/document/d/1VaYMMw30GT1me6T15qfjpsQOs8TYMcpB5r7GV8_L2p8/edit
**When** following Ingestion, Decision and Sparta steps
**Then** data should be written in Cassandra in keyspace shoppingcentertest

### CODE REVIEW

**Comment by josepablofernandez** [comment]
nice work! it's ok for me

File: plugins/src/main/scala/com/stratio/sparta/plugin/parser/ingestion/IngestionParser.scala [hunk 1]
```diff
@@ -16,82 +16,74 @@
...
-  : (List[(String, Any)], List[(String, Any)]) = {
-    if (elementList.isEmpty) {
-      (elementList, currentMap.reverse)
-    } else {
-      val currentElement = elementList.last
-      val newElementList = elementList.init
-      val newCurrentMap = currentMap ++ parseElementWithSchema(currentElement)
-      parseWithSchema(newElementList, newCurrentMap)
-    }
+@transient
```
Comment (compae) [inline plugins/src/main/scala/com/stratio/sparta/plugin/parser/ingestion/IngestionParser.scala]: Remove this?

File: plugins/src/main/scala/com/stratio/sparta/plugin/parser/ingestion/IngestionParser.scala [hunk 2]
```diff
@@ -16,82 +16,74 @@
...
+object IngestionParser extends SLF4JLogging {
+
+  val datumReader =  new SpecificDatumReader[InsertMessage](InsertMessage.getClassSchema())
+  val javaToAvro = new JavaToAvroSerializer(datumReader)
+
+  def parseRawData(rawData: Any, fieldNames: Seq[String]): Seq[Any] = {
+    Try({
```
Comment (compae) [inline plugins/src/main/scala/com/stratio/sparta/plugin/parser/ingestion/IngestionParser.scala]: Try {


}

File: plugins/src/main/scala/com/stratio/sparta/plugin/parser/ingestion/IngestionParser.scala [hunk 3]
```diff
@@ -16,82 +16,74 @@
...
+      val stratioStreamingMessage = javaToAvro.deserialize(rawData.asInstanceOf[Array[Byte]])
+
+      stratioStreamingMessage.getColumns.toList
+        // Filter to just parse those columns added to the schema
+        .filter(column => fieldNames.contains(column.getColumn))
+        .map({column: ColumnNameTypeValue => getValue(column)})}
```
Comment (compae) [inline plugins/src/main/scala/com/stratio/sparta/plugin/parser/ingestion/IngestionParser.scala]: .map{case columnNameType => getValue(columnNameType)}
}

File: plugins/src/main/scala/com/stratio/sparta/plugin/parser/ingestion/IngestionParser.scala [hunk 4]
```diff
@@ -16,82 +16,74 @@
...
-    val outputValue = outputFieldsSchema.find(field => field.name == key)
-    val transformedValue = outputValue.map( outValue =>
-      TypeOp.transformValueByTypeOp(outValue.dataType, value.asInstanceOf[Any]))
+  private def getValue(column: ColumnNameTypeValue): Any = {
+    val columnType =
+      Try(com.stratio.decision.commons.constants.ColumnType.valueOf(column.getType.toString))
+        .getOrElse(com.stratio.decision.commons.constants.ColumnType.STRING)
 
-    transformedValue.fold(List.empty[(String, Any)]) {tValue => List((key, tValue))}
+    columnType match {
```
Comment (compae) [inline plugins/src/main/scala/com/stratio/sparta/plugin/parser/ingestion/IngestionParser.scala]: Type Classes!!

File: plugins/src/test/scala/com/stratio/sparta/plugin/parser/ingestion/IngestionParserTest.scala [hunk 5]
```diff
@@ -70,16 +85,28 @@ class IngestionParserTest extends WordSpecLike with Matchers with BeforeAndAfter
...
+        1454575187944L,
+        columns.asJava,
+        new util.ArrayList[Action]())
+      val inputEvent = Row(serializeInsertMessageToAvro(insertMessage))
       val ingestionParser = new IngestionParser(ParserOrder, InputField, OutputsFields, validSchema, ParserConfig)
-      val event = ingestionParser.parse(inputEvent, false)
       val values = Seq("columnAValue", 1L, 1, 1F, 1D)
 
-      event should be eq Row.fromSeq(Seq(json) ++ event.toSeq)
+      val event = ingestionParser.parse(inputEvent, false)
```
Comment (compae) [inline plugins/src/test/scala/com/stratio/sparta/plugin/parser/ingestion/IngestionParserTest.scala]: .parse(row = inputEvent, removeRaw = false)

**Comment by compae** [comment]
Good Job!! Paga!

**Comment by danielcsant** [comment]
@compae it seems Type Classes cannot be used with Java enums...

Thanks for the review! All changes done

**Comment by compae** [comment]
LGTM

**Comment by compae** [comment]
:shipit:

**Comment by stratiocommit** [comment]
Comment accepted. Will treat it ASAP on [Jenkins](http://jenkins.stratio.com/job/Sparta/job/CD/3001/)

**Pull request merged**